### ISSUE TITLE
Lime API does not return bike_clusters anymore (and now always returns null or 50 vehicles)

### ISSUE THREAD

**Issue by hb0** [ubahnverleih/WoBike#202]
When I tested the API in February, 2021 The API behaved this way:

- `vehicles` request with zoom 16 => returns 50 `bike` entries and `bike_clusters: null`
- `vehicles` request with zoom 14 => returns `bike: null` and `bike_clusters` with entries like 0, 4, 5, 50, 100

How when I send the same requests:

- `vehicles` request with zoom 15++ => returns always 50 vehicles, never less then 50 (thus, it ignores the bounding box which is a known behavior)
- `vehicles` request with **zoom 14-- => returns both `bike: null` and `bike_clusters: null` and says in the `attributes.title: "No nearby vehicles"` as error.**

Is the error on my side or did the API behavior change?

This way it's way harder to "scrawl" a city for all it's vehicles to get an overview.

**Comment by davidwim** [comment]
Yeah, they've done some changes. I'll slightly updates the endpoints which I'll send later.

**Comment by hb0** [comment]
Cool.

So far it feels like the following parameters don't matter for the vehicle-search:
- User Location
- Size of the Bounding Box
- Zoom Level

I only get different data if I "move" the bounding Box, and, thus, the center of the BB.

I.e. **the API just seems to return 50 (closest?) vehicles around the Bounding Box **Center**** (size does not matter).

**Comment by hb0** [comment]
@W1MMER you mentions some changes to the endpoints, I'm wondering what exactly you meant there.

**Comment by W1MMER** [comment]
Oh I'm so sorry! I completely forgot about this. It was mostly the parameters from what I can remember. The old fetch bikes endpoint still works fine though.

**Comment by W1MMER** [comment]
This:

**Base URL**: `https://web-production.lime.bike/api/rider/v1/views/map`

**Parameters**:

| Parameters  | Descriptions                                                  | Mandatory |
| ------------ | ------------------------------------------| :---------:  |
| accuracy      | this: `154.2512554138973`                        | X                 |
| ne_lat            | <ne_lat>                            | X                 |
| ne_lng           | phone number intl format                            | X                 |
| sw_lat           | phone number intl format                            | X                 |
| sw_lng           | phone number intl format                            | X                 |
| user_gps_time           | phone number intl format                            | X                 |
| user_latitude           | phone number intl format                            | X                 |
| user_longitude           | phone number intl format                            | X                 |

**Comment by matt-wirtz** [comment]
I do get only 50 vehicles in the response too.

Was anyone able to circumvent this restriction? Is there a limit parameter available to override the default value of 50.

**Comment by hb0** [comment]
I didn't find anything in this direction. I just made multiple requests:
- starting with my whole area of interest
- if new vehicles / positions are found, I slice the area of interest in halb (horizontal and next time vertical)
- and repeat, until there is nothing new. Just make sure you don't send too many requests, or else your IP is restricted (not your token)

**Comment by W1MMER** [comment]
@matt-wirtz they've switched endpoints. The new one seems to be able to load more bikes but doesn't give out as much info about the bike itself.

https://web-production.lime.bike/api/rider/v2/map/bike_pins with the same params as before.

**Comment by jhZzz021** [comment]
Hello, I can only get 50 records through this URL(https://web-production.lime.bike/api/rider/v1/views/map). And get 200 records without time information through this URL (https://web-production.lime.bike/api/rider/v2/map/bike_pins ). Was anyone able to circumvent this restriction?

**Comment by ducknoir** [comment]
@W1MMER 
New here, just starting to poke around with these APIs. Can I ask, how are you learning about updates to the Lime API (e.g. the new bikepins endpoint)?  Do you have access to documentation somewhere? Sniffing traffic? I'm wanting to understand how how people figure these things out. Thanks.
